pyccl/tests/test_cclobject.py

```diff
@@ -176,6 +173,10 @@ def test_CCLObject_default_behavior():
     assert instances[0] != instances[1]
     assert hash(instances[0]) != hash(instances[1])
 
+    MyType = type("MyType", (ccl.CCLAutoreprObject,), {"test": 0})
+    instances = [MyType() for _ in range(2)]
+    assert instances[0] != instances[1]
+
```

Comment:
What is this doing?

Reply:
L176 is equivalent to
```python
class MyType(CCLAutoreprObject):
    test = 0
```
It doesn't define a `__repr_attrs__` hook so it checks that every instance's repr is Python's default which uses memory id.